src/cpp/rtps/transport/ChannelResource.h

```diff
@@ -50,6 +50,11 @@ class ChannelResource
         thread_.swap(pThread);
     }
 
+    inline eprosima::thread get_thread()
+    {
+        return std::move(thread_);
+    }
+
```

Comment:
There would be no need for this method if changing the thread is handled directly in the new `update_channel` methods